### ISSUE TITLE
Can get into the diagnose mode but no /dev/tty.usbmodem* found

### ISSUE THREAD

**Issue by zenof** [HappyZ/dpt-tools#99]
Hi,

thanks for all the works done.
I mistakenly reseted my DPT-RP1, and then when boot into system, after press home button, there is empty menu popped. Then I have tried reset several times, always empty menu. When connect to computer via USB cable, the dpt-tools also cannot make connection with device, and the official Digital Paper app can detect it but stuck on connecting.

I can reboot into diagnose mode and see the black rectangular, however, there is no any /dev/tty.usbmodem* (I have list the /dev/tty.* and /dev/cu.* but nothing related found) related found and I cannot make connection. Seems like I bricked it? Any help will appreciate!

**Comment by HappyZ** [comment]
If you are on macos, it may require you to unplug, reboot macbook, and plug back in to find it - or try a different cable. My macbook sometimes also could not show it, and it is a bug by apple.

And regarding your issue, how'd you reset? Reset in system settings should only cause the applauncher crashing..

**Comment by chenzhongli** [comment]
1、进入诊断模式后，链接电脑，电脑识别串口COM ？  用putty串口连接此设备，用户名：root 密码：12345.
2、登陆后进入目录   cd /usr/local/bin
3、输入：mass_storage 
4、电脑将会识别一个U盘
5、将下载的FwUpdater.pkg复制到U盘
6、复位DPT,再次进入诊断模式，DPT将会自动升级

**Comment by ghostwheel** [comment]
Thank you, these instructions are very useful.
On Windows 10, I did the following, after entering diagnosis mode and connecting the USB cable:
1. Open device manager (alt-x m)
2. Click on Ports (COM & LPT), when it opens you should see which serial port the DPT is connected to. Mine was COM5
3. At the prompt for /dev/tty.usbmodem, enter the port (I entered COM5).

You can also follow directions above, and use putty to connect to COM5, then enter user root, passwd 12345.

**Comment by HappyZ** [comment]
ah.. you were using windows.. make sense..

**Comment by chenzhongli** [comment]
python3 dpt-tools.py --diagnosis
对应WindowsCOM5 端口                /dev/ttyS5

**Comment by zenof** [comment]
> If you are on macos, it may require you to unplug, reboot macbook, and plug back in to find it - or try a different cable. My macbook sometimes also could not show it, and it is a bug by apple.
> 
> And regarding your issue, how'd you reset? Reset in system settings should only cause the applauncher crashing..

Thanks for reply! Yes, I am on macOS, and I tried reboot and re-plug and finally got my /dev/tty. usbmodem01. Thanks a lot!

Yes, I was reset in system settings, the applauncher is only popped a very small empty menu, like crashing.

I have successfully get into the diagnosis mode. But I have tried `factory_reset.sh`, and restore-boot-img, or install-pkg for an entire re-install. But after initial steps, the applauncher are still crashing...

How can I fix the applauncher crashing, thanks!
